nabo/nabo.cpp

```diff
@@ -46,6 +46,17 @@ namespace Nabo
 {
 	using namespace std;
 
+  struct stringformat : private stringstream {
```

Comment:
An alternative with even more potential deduplication in the code would have been to implement a special child of `runtime_error`, e.g. `nabo::runtime_error`,  that allows "shifting the message in". This would have the additional benefit of allowing client code to distinguish it from other exceptions.